### ISSUE TITLE
FTP Support

### ISSUE THREAD

**Issue by bricknerb** [PcapDotNet/Pcap.Net#34]
<b>Brickner[CodePlex]</b> <br />Support FTP protocol.

**Comment by ti24horas** [comment]
Hi,
I think the main purpose of pcapdotnet is to interpret and expose pcap packets as objects.

I've made a sip stack with udp, tcp with defragmentation support.
It's very hard and troublesome to repack thos packages.
I don't think this issue is winpcap issue.

**Comment by bricknerb** [comment]
Can you explain exactly what is troublesome?
How is it related to FTP?

**Comment by ti24horas** [comment]
When you have do "defragment" an IP packet, you should see the flag MoreFragments and read up to the next packet which does not contains "more fragments" bit set.


However when you have to deal with tcp fragmentation, you have to know how the payload is.
Ex: for sip packets, we have a status line, a list of kvp headers, optionally a body. To be able to complete the read you have to read all headers, discover the body size (from the content length header) and see if there's anoither sip packet in continuation.
For FTP (mainly ftp commands), I think there's no problem because generally you have line-by-line protocol

**Comment by bricknerb** [comment]
It seems like the issue you're discussing here is more related to https://github.com/PcapDotNet/Pcap.Net/issues/19, no?

**Comment by ti24horas** [comment]
@bricknerb yes, related to issue #19, however, there are no generic solution as it depends on protocol.

**Comment by bricknerb** [comment]
No, TCP reconstruction doesn't depend on any protocol except IP and TCP.
